### PR TITLE
feat(utility): add FileBase, File and RemoteFile for basic binary files

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2mez3rv 实现基础 File 类型，支持提取文件后缀 + 文件大小](https://app.clickup.com/t/2mez3rv)

File: graviti/utility/binary_file.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        """Get the key of the file.
+
+        Returns:
+            The key of the file.
+
+        """
+        return self._key
+
+    @property
+    def format(self) -> str:
```
Comment (linjiX) [inline graviti/utility/binary_file.py:55]: should format be `Optional`

File: graviti/utility/binary_file.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+        Returns:
+            The File or RemoteFile instance.
+
+        """
+        key = scalar["key"].as_py()
+        format_: str = scalar["format"].as_py()
+        size: int = scalar["size"].as_py()
+
+        if key.startswith("file://"):
```
Comment (linjiX) [inline graviti/utility/binary_file.py:90]: the key of remote file is not starts with `oss://` or `s3://`, is it a good choice to add `file://` suffix for local file?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/utility/binary_file.py (hunk 2 above)
Comment (Lee-000) [reply graviti/utility/binary_file.py:90]: But we need to distinguish the local file and the remote file with the key.

File: graviti/utility/binary_file.py (hunk 1 above)
Comment (Lee-000) [reply graviti/utility/binary_file.py:55]: The format could be an empty string.

Comment (linjiX) [reply graviti/utility/binary_file.py:55]: None or empty string, the schema says the format is nullable?

File: graviti/utility/binary_file.py (hunk 2 above)
Comment (linjiX) [reply graviti/utility/binary_file.py:90]: we need a better way to distinguish the local and remote file.

File: graviti/utility/binary_file.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    def size(self) -> int:
+        """Get the size of the file.
+
+        Returns:
+            The size of the file.
+
+        """
+        return self._size
+
+    def to_post_data(self) -> Dict[str, Union[int, str]]:
```
Comment (linjiX) [inline graviti/utility/binary_file.py:64]: should this be private?

File: graviti/utility/binary_file.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    @property
+    def key(self) -> str:
+        """Get the key of the file.
+
+        Returns:
+            The key of the file.
+
+        """
+        if not hasattr(self, "_key"):
+            self._key = self._path.as_uri()
```
Comment (linjiX) [inline graviti/utility/binary_file.py:122]: why the key is `uri`?

File: graviti/utility/binary_file.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+        Returns:
+            The extension of the file.
+
+        """
+        if not hasattr(self, "_extension"):
+            suffix = self._path.suffix
+            if suffix in _ENCODINGS:
+                suffix = Path(self._path.stem).suffix + suffix
+            self._extension = suffix.lstrip(".")
```
Comment (linjiX) [inline graviti/utility/binary_file.py:137]: will it be better to keep the leading `.`

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/utility/binary_file.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    @property
+    def key(self) -> str:
+        """Get the key of the file.
+
+        Returns:
+            The key of the file.
+
+        """
+        if not hasattr(self, "_key"):
+            self._key = self._path.as_posix()
```
Comment (linjiX) [inline graviti/utility/binary_file.py:116]: why using `as_posix`?

**Review by linjiX — APPROVED** [review]

**Pull request merged**